src/main/scala/org/opencypher/memcypher/api/MemCypherRecords.scala

```diff
@@ -79,6 +79,13 @@ case class Embeddings(data: List[CypherMap]) {
   def select(fields: Seq[String])(implicit header: RecordHeader, context: MemRuntimeContext): Embeddings =
     copy(data = data.map(row => row.filterKeys(fields)))
 
+  def distinct(fields: Set[Var])(implicit header: RecordHeader, context: MemRuntimeContext): Embeddings = {
+    // converting "." to "_dot_" to match with header
+    val columnNames = for{ x <- fields} yield x.name.replace(".","_dot_")
```

Comment:
There is already a helper function for these conversions: look for `ColumName.of`

Also, just a little style hint. Instead of the for comprehension you could `fields.map(ColumName.of)`